src/domain_chunk.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018-2021 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "rmutil/alloc.h"
+#include "domain_chunk.h"
+
+static inline void _init_domain_chunk(DomainChunk *chunk) {
+    chunk->num_samples = 0;
+    chunk->rev = false;
+    chunk->size = 0;
+    chunk->samples.timestamps = NULL;
+    chunk->samples.values = NULL;
+}
+
+DomainChunk *allocateDomainChunk() {
+    DomainChunk *chunk = (DomainChunk *)malloc(sizeof(DomainChunk));
+    _init_domain_chunk(chunk);
+    return chunk;
+}
+
+void ReallocDomainChunk(DomainChunk *chunk, size_t n_samples) {
+    _init_domain_chunk(chunk);
```

Comment:
this is a bit confusing, usually "New"/Init function would allocate and initialize everything and realloc should be able to grow the "container" accordingly.